atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/SchemaMutationLockTables.java

```diff
@@ -63,31 +66,61 @@ public SchemaMutationLockTables(CassandraClientPool clientPool, CassandraKeyValu
     }
 
     public TableReference createLockTable() throws TException {
-        return clientPool.runWithRetry(this::createLockTable);
-    }
-
-    private TableReference createLockTable(CassandraClient client) throws TException {
-        String lockTableName = LOCK_TABLE_PREFIX + "_" + getUniqueSuffix();
-        TableReference lockTable = TableReference.createWithEmptyNamespace(lockTableName);
+        TableReference lockTable = TableReference.createWithEmptyNamespace(LOCK_TABLE_PREFIX);
         log.info("Creating lock table {}", SafeArg.of("schemaMutationTableName", lockTable));
-        createTableInternal(client, lockTable);
+        createTableWithCustomId(lockTable);
         return lockTable;
     }
 
-    private String getUniqueSuffix() {
-        // We replace '-' with '_' as hyphens are forbidden in Cassandra table names.
-        return UUID.randomUUID().toString().replace('-', '_');
-    }
+    private void createTableWithCustomId(TableReference tableRef) throws TException {
+        String internalTableName = CassandraKeyValueServiceImpl.internalTableName(tableRef);
+        String keyspace = config.getKeyspaceOrThrow();
+        String fullTableNameForUuid = keyspace + "." + internalTableName;
+        UUID uuid = UUID.nameUUIDFromBytes(fullTableNameForUuid.getBytes());
+
+        // TODO use parameters as in ColumnFamilyDefinitions.getStandardCfDef
```

Comment:
I suggest making this simpler by having the method be very schema mutation lock table specific so you don't have to worry about arbitrary column family definitions